lib/bindConnectors/interceptAfterSuccess.js

```diff
@@ -7,12 +8,22 @@ module.exports = function (message) {
 
 		message.model.afterSuccess = function (body, params, res) {
 
-			body = ( _.isArray(body) ? { results: body } : body );
+			return when.promise(function (resolve, reject) {
 
-			return ( modelAfterSuccess ? modelAfterSuccess(body, params, res) : body );
+				when(
+					( modelAfterSuccess ? modelAfterSuccess(body, params, res) : body )
+				)
+
+				.then(function (body) {
```

Comment:
This looks good, the only thing I'm slightly conscious of here is something that is a bit of an edge-case: `afterSuccess` doesn't necessarily have to return anything - it can just mutate the data instead. See https://github.com/trayio/threadneedle/blob/master/lib/addMethod/globalize/afterSuccess.js#L26.

The cases that would affect this function are pretty rare, but for safety, I'd recommend something more like this:

```js
.then(function (modifiedBody) {
  if (_.isArray(modifiedBody)) {
    return { results: modifiedBody };
  } else if (_.isArray(body)) {
    return { results: body };
  } else {
    return body;
  }
})
```